Source/WebCore/css/calc/CSSCalcOperationNode.h

```diff
@@ -145,7 +145,7 @@ class CSSCalcOperationNode final : public CSSCalcExpressionNode {
     }
 
     void makeTopLevelCalc();
-    bool shouldNotPreserveFunction() const { return isMinOrMaxNode() || isHypotNode(); }
+    bool shouldNotPreserveFunction() const { return isSignNode() || isMinOrMaxNode() || isExpNode() || isHypotNode() || isPowOrSqrtNode(); }
```

Comment:
Could we define `shouldNotPreserveFunction` in terms of `shouldPreserveFunction` or vice versa? Seems a shame to express the same list twice.

Reply:
We probably could, but `shouldPreserveFunction` will get removed anyway as I fix the serialization for the remaining functions since it's actually undesirable behavior.

I'd stay on the side of caution here and keep it that way for now (and have the individual changes for the other functions slowly remove `shouldPreserveFunction`).

If we really wanted, it would probably look something like:
```
shouldPreserveFunction() { return !shouldNotPreserveFunction() && !isCalcSumNode() && !isCalcProductNode() && m_operator != CalcOperator::Subtract && m_operator != CalcOperator::Divide && m_operator != CalcOperator::Multiply;
```

Reply:
We can wait and look at it after you are done with the individual fixes.